File: uta-repo/snippets/java/Verify.java
// UTA — Java example: verify a credential via public HTTP API (JDK 11+).
// Repo: https://github.com/alicelabs-llc/universal-trust-adapter
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;

public class Verify {
    public static void main(String[] args) throws Exception {
        if (args.length < 1) { System.err.println("Usage: java Verify <card>"); System.exit(1); }
        String card = args[0];
        String payload = String.format("{\"card\":\"%s\"}", card);
        HttpClient client = HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(5)).build();
        HttpRequest req = HttpRequest.newBuilder()
            .uri(URI.create("https://www.marketnow.site/api/trust?action=verify"))
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(payload))
            .build();
        HttpResponse<String> resp = client.send(req, HttpResponse.BodyHandlers.ofString());
        System.out.println(resp.body());
    }
}

File: uta-repo/snippets/kotlin/Verify.kt
// UTA — Kotlin example: verify a credential via public HTTP API (Ktor).
// Repo: https://github.com/alicelabs-llc/universal-trust-adapter
import io.ktor.client.*
import io.ktor.client.call.*
import io.ktor.client.request.*
import io.ktor.client.statement.*
import io.ktor.http.*
import kotlinx.coroutines.runBlocking
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.jsonObject
import kotlinx.serialization.json.jsonPrimitive

fun main(args: Array<String>) = runBlocking {
    val card = if (args.isNotEmpty()) args[0] else error("Usage: verify <card>")
    val client = HttpClient()
    val response: HttpResponse = client.post("https://www.marketnow.site/api/trust") {
        parameter("action", "verify")
        contentType(ContentType.Application.Json)
        setBody("""{"card":"$card"}""")
    }
    val body: String = response.body()
    val json = Json.parseToJsonElement(body).jsonObject
    println("Decision:   ${json["decision"]?.jsonPrimitive()?.content}")
    println("Format:     ${json["detected_format"]?.jsonPrimitive()?.content}")
    println("Issuer:     ${json["issuer"]?.jsonPrimitive()?.content}")
    client.close()
}
